src/app/reminder-list/page.tsx

```diff
@@ -1,86 +1,207 @@
-        {/* main */}
-        <Header
-          title={showHeader}
-          type="primary"
-          iconLeft={
-            <IconButton aria-label="back" className="bg-base-100">
-              <ChevronLeftIcon
-                fontSize="large"
-                sx={{ padding: 1 }}
-                onClick={handleBackButtonClick}
-              />
-            </IconButton>
-          }
-          iconRight={
-            showAddButton ? (
-              <Button>Add</Button>
-            ) : (
+      {editing ? (
+        <div className="flex h-screen flex-col">
+          <Header
+            title="Edit Reminder"
+            type="primary"
+            iconLeft={
               <IconButton aria-label="back" className="bg-base-100">
-                <AddIcon
-                  fontSize="large"
-                  onClick={handleAddButtonClick}
-                  sx={{ padding: 1 }}
-                />{" "}
+                <LuStepBack
+                  className="text-primary"
+                  size={30}
+                  strokeWidth={1.3}
+                  onClick={handleCancelClick}
+                />
               </IconButton>
-            )
-          }
-        />
-
-        <div className="w-full flex-1 overflow-y-auto bg-base-100">
-          {showForm ? (
-            <div className="  flex flex-col gap-4 px-6 py-2">
-              <AddReminer />
+            }
+            iconRight={
+              <IconButton aria-label="back" className="bg-base-100">
+                <LuSave
+                  className="text-primary"
+                  size={30}
+                  strokeWidth={1.3}
+                  onClick={handleSaveClick}
+                />
+              </IconButton>
+            }
+          />
+          <AddReminder />
+          <Footer />
+        </div>
+      ) : (
+        <div className="flex h-screen flex-col">
+          <Header
+            type="primary"
+            iconLeft={
+              <LuSearch size={30} strokeWidth={1.3} className="text-primary" />
+            }
+            iconRight={
+              <LuPlusCircle
+                size={30}
+                strokeWidth={1.3}
+                className="text-primary"
+                onClick={handleAddNewButton}
+              />
+            }
+            title="Reminders"
+          />
+          <div className="flex w-full flex-1 flex-col overflow-hidden bg-base-100">
+            {/* main */}
+            <div className="flex-1 overflow-y-auto p-4">
+              {(() => {
+                const reminderCards = [];
+                for (let i = 0; i < reminders.length; i++) {
+                  reminderCards.push(
+                    <div
+                      className="relative mb-4 flex cursor-pointer items-center rounded-lg bg-white p-4 shadow"
+                      onClick={() => handleReminderCardClick(i)}
+                      onKeyDown={() => handleReminderCardClick(i)}
+                    >
```

Comment:
Missing `key` prop. 

React requires a unique `key` prop for each child in a list when rendering multiple components. This helps React identify which items have changed, are added, or are removed, and improves performance by allowing React to only re-render the components that actually changed.